MobileBuy/sample/src/main/java/com/shopify/sample/BaseApplication.java

```diff
@@ -51,21 +52,37 @@
 import io.reactivex.android.schedulers.AndroidSchedulers;
 import timber.log.Timber;
 
-import static java.util.Collections.emptyList;
 import static java.util.Collections.emptySet;
 
 public abstract class BaseApplication extends Application {
+
+  private static BaseApplication instance;
+
   private final FrescoMemoryTrimmableRegistry frescoMemoryTrimmableRegistry = new FrescoMemoryTrimmableRegistry();
   private MutableLiveData<ShopSettings> shopSettings = new MutableLiveData<>();
+  private UseCases useCases;
+
+  public static BaseApplication getInstance() {
+    return instance;
+  }
 
   @Override
   public void onCreate() {
     super.onCreate();
+
+    instance = this;
+    initialize();
+    useCases = createUseCases();
+
     iniTimber();
     initFresco();
     fetchShopSettings();
   }
 
+  public UseCases getUseCases() {
```

Comment:
UseCases are singleton for simplification. It is better to use Dagger but it is not necessary in sample app.